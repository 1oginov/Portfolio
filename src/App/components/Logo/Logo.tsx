import { Typography } from '@material-ui/core';
import { BusinessCenterOutlined } from '@material-ui/icons';
import * as React from 'react';

interface Props {
  className?: string;
}

const Logo: React.FunctionComponent<Props> = ({ className }: Props) => (
  <Typography className={className} variant="h4">
    <BusinessCenterOutlined />
    {' '}
    Portfolio
  </Typography>
);

export default Logo;
