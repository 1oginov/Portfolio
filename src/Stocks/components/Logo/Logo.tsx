import * as React from 'react';

// TODO: Tests.

interface Props {
  url: string | null;
}

const Logo: React.FunctionComponent<Props> = ({ url }: Props) => {
  if (!url) {
    return null;
  }

  return <img alt="" src={url} style={{ height: 32, objectFit: 'contain', width: 32 }} />;
};

export default Logo;
