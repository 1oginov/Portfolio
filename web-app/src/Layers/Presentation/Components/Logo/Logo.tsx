import { Typography } from '@material-ui/core';
import { BusinessCenterOutlined } from '@material-ui/icons';
import * as React from 'react';

interface Props {
  className?: string;
}

export const Logo: React.FunctionComponent<Props> = ({ className }: Props) => (
  <Typography className={className} variant="h4">
    <BusinessCenterOutlined />
    {' '}
    Portfolio
  </Typography>
);

Logo.defaultProps = {
  className: undefined,
};
