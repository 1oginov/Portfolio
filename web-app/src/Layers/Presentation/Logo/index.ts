import { Logo as LogoComponent } from './Logo';

export const Logo = LogoComponent;
