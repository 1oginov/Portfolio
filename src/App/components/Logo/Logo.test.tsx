/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import Logo from './Logo';

it('matches snapshot', () => {
  const wrapper = shallow(<Logo />);

  expect(wrapper).toMatchSnapshot();
});
