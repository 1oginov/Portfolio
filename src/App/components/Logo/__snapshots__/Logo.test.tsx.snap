// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<WithStyles(ForwardRef(Typography))
  variant="h4"
>
  <BusinessCenterOutlinedIcon />
   
  Portfolio
</WithStyles(ForwardRef(Typography))>
`;
